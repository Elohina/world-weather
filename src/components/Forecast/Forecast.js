import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import styled from 'styled-components';

const StyledForecast = styled.div`
  width: 50%;
  padding: 3em;

  @media(max-width: 75em) {
    width: 100%;
  }
`;

const Forecast = ({ forecast, onChangeTime, isToday, selectedDay }) => { 
  const [selectedCard, setSelectedCard] = useState(null);

  useEffect(() => {
    if (selectedDay[0]) {
      setSelectedCard(selectedDay[0].date_label);
    }
  }, [selectedDay]);

  const onSelectCard = (data, dateLabel) => {
    onChangeTime(data, dateLabel);
    setSelectedCard(dateLabel);
  }

  return (
    <StyledForecast>
      <h3>Week</h3>
      <h5>Click on each day to see the details</h5>
      { Object.keys(forecast).map((day, i) => {
        return (
          <Card
            key={i}
            data={forecast[day][0]}
            onSelectCard={(data, dateLabel) => (onSelectCard(data, dateLabel))}
            dateLabel={forecast[day][0].date_label}
            isToday={isToday}
            isSelected={selectedCard === forecast[day][0].date_label}
          />
        );
      })}
    </StyledForecast>
  );
};

export default Forecast;

Forecast.propTypes = {
  forecast: PropTypes.object,
  onChangeTime: PropTypes.func,
  isToday: PropTypes.func
};
