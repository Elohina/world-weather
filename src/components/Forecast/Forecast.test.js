import React from "react";
import { shallow } from "enzyme";
import Forecast from "./Forecast";

const forecast = {
  16: [
    {
      dt: 1541430000,
      main: {
        temp: 284.37,
        temp_min: 284.37,
        temp_max: 284.673,
        pressure: 1008.66,
        sea_level: 1013.28,
        grnd_level: 1008.66,
        humidity: 98,
        temp_kf: -0.3
      },
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10d"
        }
      ],
      clouds: {
        all: 92
      },
      wind: {
        speed: 6.42,
        deg: 146.004
      },
      rain: {
        "3h": 0.0325
      },
      sys: {
        pod: "d"
      },
      dt_txt: "2018-11-05 15:00:00",
      date_label: 16
    }
  ],
  17: [
    {
      dt: 1541430000,
      main: {
        temp: 284.37,
        temp_min: 284.37,
        temp_max: 284.673,
        pressure: 1008.66,
        sea_level: 1013.28,
        grnd_level: 1008.66,
        humidity: 98,
        temp_kf: -0.3
      },
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10d"
        }
      ],
      clouds: {
        all: 92
      },
      wind: {
        speed: 6.42,
        deg: 146.004
      },
      rain: {
        "3h": 0.0325
      },
      sys: {
        pod: "d"
      },
      dt_txt: "2018-11-05 15:00:00",
      date_label: 16
    }
  ]
};


const onChangeTime = jest.fn();
const isToday = jest.fn();

describe("<Forecast />", () => {
  const wrapper = shallow(<Forecast forecast={forecast} onChangeTime={onChangeTime} isToday={isToday} selectedDay={forecast[0]}/>);

  it("should render component correctly", () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('Card').length).toBe(2);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
