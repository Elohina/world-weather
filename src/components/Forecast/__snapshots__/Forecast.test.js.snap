// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forecast /> should match the snapshot 1`] = `"<div class=\\"sc-AxmLO dzSCew\\"><h3>Week</h3><h5>Click on each day to see the details</h5><div id=\\"1541430000heading\\" class=\\"sc-AxiKw eWlrAk\\"><div style=\\"align-items:start\\" class=\\"sc-AxhCb eeHLGY\\"><h4 class=\\"sc-AxjAm guIcNj\\">Monday</h4><h5 class=\\"sc-AxirZ cGfImr\\">5 Nov 18</h5></div><div class=\\"sc-AxhCb eeHLGY\\"><img src=\\"http://openweathermap.org/img/wn/10d@2x.png\\" alt=\\"\\"/><p class=\\"sc-AxgMl gNhA-Dl\\">Rain</p></div><div class=\\"sc-AxhCb eeHLGY\\"><p class=\\"sc-AxhUy iIDvKs\\">92%</p><p class=\\"sc-AxgMl gNhA-Dl\\">Clouds</p></div><div class=\\"sc-AxhCb eeHLGY\\"><p class=\\"sc-AxhUy sc-AxheI kazPFF\\">284.37째C</p><p class=\\"sc-AxhUy sc-Axmtr jcEKiY\\">284.673째C</p></div></div><div id=\\"1541430000heading\\" class=\\"sc-AxiKw eWlrAk\\"><div style=\\"align-items:start\\" class=\\"sc-AxhCb eeHLGY\\"><h4 class=\\"sc-AxjAm guIcNj\\">Monday</h4><h5 class=\\"sc-AxirZ cGfImr\\">5 Nov 18</h5></div><div class=\\"sc-AxhCb eeHLGY\\"><img src=\\"http://openweathermap.org/img/wn/10d@2x.png\\" alt=\\"\\"/><p class=\\"sc-AxgMl gNhA-Dl\\">Rain</p></div><div class=\\"sc-AxhCb eeHLGY\\"><p class=\\"sc-AxhUy iIDvKs\\">92%</p><p class=\\"sc-AxgMl gNhA-Dl\\">Clouds</p></div><div class=\\"sc-AxhCb eeHLGY\\"><p class=\\"sc-AxhUy sc-AxheI kazPFF\\">284.37째C</p><p class=\\"sc-AxhUy sc-Axmtr jcEKiY\\">284.673째C</p></div></div></div>"`;
